packages/react-native-windows-init/src/Cli.ts

```diff
@@ -0,0 +1,291 @@
+    describe: 'Overwrite any existing files without prompting',
+  },
+}).argv;
+
+const EXITCODE_NO_MATCHING_RNW = 2;
+const EXITCODE_UNSUPPORTED_VERION_RN = 3;
+const EXITCODE_USER_CANCEL = 4;
+
+function REACT_NATIVE_WINDOWS_GENERATE_PATH() {
+  return require.resolve('react-native-windows/local-cli/generate-windows.js', {
+    paths: [process.cwd()],
+  });
+}
+
+function getReactNativeAppName() {
+  console.log('Reading application name from package.json...');
+  let name = JSON.parse(fs.readFileSync('package.json', 'utf8')).name;
+  if (!name) {
+    if (fs.existsSync('app.json')) {
+      console.log('Reading application name from app.json...');
+      name = JSON.parse(fs.readFileSync('app.json', 'utf8')).name;
+    }
+  }
+  if (!name) {
+    console.error('Please specify name in package.json or app.json');
+  }
+  return name;
+}
+
+function getReactNativeVersion() {
+  console.log('Reading react-native version from node_modules...');
+  const rnPkgJsonPath = require.resolve('react-native/package.json', {
+    paths: [process.cwd()],
+  });
+  if (fs.existsSync(rnPkgJsonPath)) {
+    return require(rnPkgJsonPath).version;
+  }
+
+  console.error(
+    'Error: Must be run from a project that already depends on react-native, and has react-native installed.',
+  );
+  process.exit(1);
+}
+
+function ErrorOutOnUnsupportedVersionOfReactNative(rnVersion: string) {
+  console.error(`Error: Unsupported version of react-native: ${chalk.cyan(
+    rnVersion,
+  )}
+react-native-windows supports react-native versions ${chalk.cyan('>=0.60')}`);
+  process.exit(EXITCODE_UNSUPPORTED_VERION_RN);
+}
+
+function getDefaultReactNativeWindowsSemVerForReactNativeVersion(
+  rnVersion: string,
+) {
+  const validVersion = semver.valid(rnVersion);
+  if (validVersion) {
+    const major = semver.major(validVersion);
+    const minor = semver.minor(validVersion);
+    if (major === 0 && minor >= 60) {
+      return `^${major}.${minor}.0-0`;
+    }
+  }
+  ErrorOutOnUnsupportedVersionOfReactNative(rnVersion);
+}
+
+function getMatchingReactNativeSemVerForReactNativeWindowsVersion(
+  rnwVersion: string,
+) {
+  const validVersion = semver.valid(rnwVersion);
+  if (validVersion) {
+    const major = semver.major(validVersion);
+    const minor = semver.minor(validVersion);
+    if (major === 0 && minor >= 59) {
+      return `^${major}.${minor}`;
+    }
+  }
+  return 'unknown';
+}
+
+function getLatestMatchingVersion(
+  pkg: string,
+  versionSemVer: string,
+): Promise<string> {
+  return new Promise((resolve, reject) => {
+    if (semver.validRange(versionSemVer)) {
+      npm.packages.releases(
+        pkg,
+        (err: any, details: {[key: string]: object}) => {
+          if (err) {
+            reject(err);
+          } else if (details) {
+            const versions = Object.keys(details);
+            if (versions.length > 0) {
+              const candidates = versions
+                .filter(_ => semver.satisfies(_, versionSemVer))
```

Comment:
We shouldn't need this logic at all if we have a real semver to follow. We can call `npm.packages.range` with the semver and it will give us something that works. The logic here was previously needed because we were injecting custom prerelease tags into the process, so we did our own matching, but this can be removed now that we're vnext only.

Reply:
Also though, underscore usually means to discard the param. We shouldn't use it if we're using it.

Reply:
Oh, forgot that TS uses _ as a special thing.  Bad habit from C# where it was the standard local var name for trivial lambdas

Reply:
Unfortunately `npm.packages.range` doesn't do what we need it to do.  `npm.packages.range('react-native-windows', '^0.61.0-0') returns no results.  When it should return all the pre-release versions.

Reply:
Could you leave a comment about that behavior?